### PR TITLE
[JENKINS-40997] Add way to get locked resource name in pipeline

### PR DESCRIPTION
**Pull request by f-julian**
An configurable environment variable is now set with the name of the locked resource fixes JENKINS-40997

### CODE REVIEW

**Comment by f-julian** [comment]
An configurable environment variable is now set with the name of the locked resource fixes JENKINS-40997

**Comment by subos2008** [comment]
Is this likely to go in? We're currently running with this manually patched in.

**Comment by Senorsen** [comment]
subos2008 +1, really hope this can be merged into the future release.

File: src/main/java/org/jenkins/plugins/lockableresources/LockStep.java [hunk 1]
```diff
@@ -98,18 +106,33 @@ public static FormValidation doCheckResource(@QueryParameter String value, @Quer
...
+
+	    StringBuilder result = new StringBuilder();
+
 		// a label takes always priority
 		if (this.label != null) {
+		    result.append("Label: ").append(this.label);
+
 			if (this.quantity > 0) {
-				return "Label: " + this.label + ", Quantity: " + this.quantity;
+                result.append(", Quantity: ").append(this.quantity);
```
Comment (drewish) [inline src/main/java/org/jenkins/plugins/lockableresources/LockStep.java]: indenting seems wonky here

File: src/main/java/org/jenkins/plugins/lockableresources/LockStep.java [hunk 2]
```diff
@@ -98,18 +106,33 @@ public static FormValidation doCheckResource(@QueryParameter String value, @Quer
...
 			}
-			return "Label: " + this.label;
+
+			if (variable != null) {
+			    result.append(", Variable: ").append(this.variable);
+            }
```
Comment (drewish) [inline src/main/java/org/jenkins/plugins/lockableresources/LockStep.java]: Looks like spaces rather than tabs like the rest of the file is using?

**Comment by f-julian** [comment]
Hi thanks for the review. Changed to tabs for the whole file.

**Comment by pbronneberg** [comment]
Great additional feature. Any update on when we can expect this to be merged? Anything I can do to help?

**Comment by subos2008** [comment]
+1 please can we have this merged!

**Comment by dgeissl** [comment]
@drewish can we expect to see the changes anytime soon to be integrated?

**Comment by drewish** [comment]
@dgeissl I'm not a maintainer. I was just pointing out how the changes differed from the existing style.

**Comment by dgeissl** [comment]
Sorry about the missunderstanding, I just saw the review tag on you.

**Comment by f-julian** [comment]
Hi i think @amuniz and @abayer are the maintainers for this plugin i will try get in touch with them.
If someone could do another code review or test that would help to.

File: src/main/java/org/jenkins/plugins/lockableresources/LockStepExecution.java [hunk 3]
```diff
@@ -90,6 +114,28 @@ protected void finished(StepContext context) throws Exception {
...
+
+        private ResourceVariableNameExpander(String resourceVariableName, List<String> resourceNames) {
+            this.overrides = new HashMap<>();
+            this.overrides.put(resourceVariableName, joinResourceNames(resourceNames));
+        }
+
+        private String joinResourceNames(List<String> resourceNames) {
+            StringBuilder resourceName = new StringBuilder();
+            for (String res : resourceNames) {
+                resourceName.append((resourceName.length() == 0) ? res : ", " + res);
```
Comment (abishek3876) [inline src/main/java/org/jenkins/plugins/lockableresources/LockStepExecution.java]: The delimiter should either be a comma or a space, not both. We should also make sure the delimiter cannot be used in the resource name.

**Comment by abishek3876** [comment]
@f-julian were you able to get in touch with the maintainer? Any chance this could end up in a release in near future?

**Comment by subos2008** [comment]
You can also apply to become a maintainer yourself if the current ones are busy. Via the jenkins dev email list iirc.

**Comment by amuniz** [comment]
Isn't https://github.com/jenkinsci/lockable-resources-plugin/pull/49 achieving the same in more concise way?

**Comment by sheepdreamofandroids** [comment]
@amuniz Yes, it does. Please just pick one and merge it so we can stop running patched versions of the plugin.

**Comment by f-julian** [comment]
@amuniz Yes #49 is the same feature.
IMHO #50 has more tests (resource var in nested, parallel blocks) and the changes for the snipped generator.

But yes please pick one.

**Comment by scott-zhong** [comment]
Hi, this is something we need, we're running with this manually patched in.  Please select #49 or #50 for inclusion into master.

**Comment by koan00** [comment]
@amuniz and @abayer - Any more feedback? There appears to be a lot of interest in this feature.

**Comment by tite-alvarado** [comment]
Hello, this awesome plugin already does a great job for us.
I'm also looking to get this feature released to be able to deal with parallelizing some pipeline stages.

**Comment by amuniz** [comment]
Just merged #49 which includes exactly this feature.
@f-julian is it ok if I close this one?

**Comment by f-julian** [comment]
@amuniz That is totally ok for me.

Closed in favor of [#49](https://github.com/jenkinsci/lockable-resources-plugin/pull/49)

**Comment by anthuwan13** [comment]
Should I wait for this feature to be available in the plugins for the next release ? I don't just want to install the patch

**Pull request closed without merging**